lib/auth.js

```diff
@@ -12,11 +12,14 @@ const urls = require('./urls');
 const { generateBasicAuthToken } = require('./utils/auth');
 
 /**
- * Token lock status
+ * Token lock status for v4 and PII
  *
  * @private
  */
-let tokenLocked = false;
+const tokenLockStatus = {
```

Comment:
Covert to `new Map([['v4', false], ['pii', false])`?

Reply:
Why a `Map`?